internal/cmd/query/formatting_printer.go

```diff
@@ -0,0 +1,269 @@
+package query
+
+import (
+	"encoding/base64"
+	"io"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/influx-cli/v2/pkg/fluxcsv"
+)
+
+// formattingPrinter formats query results into a structured table before printing.
+type formattingPrinter struct {
+	widths    []int
+	maxWidth  int
+	newWidths []int
+	pad       []byte
+	dash      []byte
+	// fmtBuf is used to format values
+	fmtBuf [64]byte
+
+	cols       []fluxcsv.FluxColumn
+	lastColIdx int
+}
+
+func NewFormattingPrinter() *formattingPrinter {
+	return &formattingPrinter{}
+}
+
+func (f *formattingPrinter) PrintQueryResults(resultStream io.ReadCloser, out io.Writer) error {
+	res := fluxcsv.NewQueryTableResult(resultStream)
+	defer res.Close()
+	return f.write(res, out)
+}
+
+const fixedWidthTimeFmt = "2006-01-02T15:04:05.000000000Z"
+
+var eol = []byte{'\n'}
+
+type writeHelper struct {
+	w   io.Writer
+	err error
+}
+
+func (w *writeHelper) write(data []byte) {
+	if w.err != nil {
+		return
+	}
+	_, err := w.w.Write(data)
+	w.err = err
+}
+
+var minWidthsByType = map[fluxcsv.ColType]int{
+	fluxcsv.BoolDatatype:         12,
+	fluxcsv.LongDatatype:         26,
+	fluxcsv.ULongDatatype:        27,
+	fluxcsv.DoubleDatatype:       28,
+	fluxcsv.StringDatatype:       22,
```

Comment:
General comment/question...I know these min widths are consistent with the existing CLI but some of them seem a little excessive. Like for a string, it seems like it ends up with a lot of wasted space for many strings. I'm not necessarily suggesting you change the formatting here, but I am curious if these min widths have some significance outside of appearance?

Reply:
I could image some ideal future where the column min widths are based on the actual terminal size of the user, but that's probably a battle for another day :smile:

Reply:
As far as I can tell the only min width that makes sense here is the one for time columns. Detecting based on terminal width does sound very tempting... it's very annoying to use `influx query` as-is when I've got it on my vertical monitor.

I'll make a ticket!